### PR TITLE
Removing Array.newInstance calls

### PR DESCRIPTION
**Pull request by satybald**
`Array.newInstance` is java reflection API, which quite slow in practice and there is no real need for them. 

> Reflection is powerful, but should not be used indiscriminately. If it is possible to perform an operation without using reflection, then it is preferable to avoid using 

https://docs.oracle.com/javase/tutorial/reflect/

### CODE REVIEW

**Comment by satybald** [comment]
build failed ;( could you please give the logs? @hjafarpour @dguy @apurvam @rodesai

**Comment by apurvam** [comment]
Looks like a bunch of check style failures. Do those not reproduce locally?

**Comment by satybald** [comment]
@apurvam nope, I'm not able to run verify maven locally due to confluent maven issue. 

```
[ERROR] Failed to execute goal on project ksql-engine: Could not resolve dependencies for project io.confluent.ksql:ksql-engine:jar:4.1.0-SNAPSHOT: Could not find artifact org.apache.kafka:kafka-streams-test-utils:jar:1.1.0-SNAPSHOT in confluent (http://packages.confluent.io/maven/) -> [Help 1]
```

**Comment by apurvam** [comment]
Ah. I'm sorry. Will follow up to see what's wrong with the public build.

File: ksql-engine/src/main/java/io/confluent/ksql/function/udaf/topkdistinct/TopkDistinctKudaf.java [hunk 1]
```diff
@@ -33,23 +32,21 @@
 import io.confluent.ksql.util.KsqlException;
 
 public class TopkDistinctKudaf<T> extends KsqlAggregateFunction<T, T[]> {
-
-  private Integer tkVal;
-  private T[] tempTopkArray;
-  private Class<T> ttClass;
+  private final int topKSize;
```
Comment (dguy) [inline ksql-engine/src/main/java/io/confluent/ksql/function/udaf/topkdistinct/TopkDistinctKudaf.java]: I don't understand these changes as this has already been done: https://github.com/confluentinc/ksql/blob/master/ksql-engine/src/main/java/io/confluent/ksql/function/udaf/topk/TopkKudaf.java#L33

**Comment by satybald** [comment]
@apurvam I created an issue for this #727

**Comment by apurvam** [comment]
Thanks, we are working on fixing it.

**Comment by apurvam** [comment]
@satybald  you should be able to build locally and reproduce the checkstyle failures. Thanks!

**Comment by satybald** [comment]
@big-andy-coates I don't mind closing the PR. However, I would personally love to see results of benchmarks between `new Object` and `Arrays.newInstance`.

**Comment by big-andy-coates** [comment]
@satybald the testing I alluded to was done a while ago and was hardly conclusive - just a simple test that created arrays of different sizes, left running for many minutes and timed.  So, easy enough for anyone to recreate locally, but not a true x-platform performance test! 

However, for this PR to switch from one to the other, there would need to be strong evidence the change offered a significant performance improvement, considering the loss of type information.

**Pull request closed without merging**